otherlibs/systhreads/st_stubs.c

```diff
@@ -456,6 +462,7 @@ static void caml_thread_domain_initialize_hook(void)
 
   new_thread->domain_id = Caml_state->id;
   new_thread->descr = caml_thread_new_descriptor(Val_unit);
+  new_thread->dls_state = Atom(0);
```

Comment:
Why is this `Atom(0)`? Does `Val_unit` not work?

Reply:
Earlier, this initial value was accessed. In the new version, it is no longer accessed.

Except, if an async callback accesses the TLS before it is initialized. This would be nonsense but it should not crash the program. As it turns out, the initial value of `Caml_state->dls_root = Val_unit` in the current implementation would cause a crash in this situation, so it is in fact better to replace it with `Atom(0)` in `domain.c` too.

(This moving of initialization to the C side because of hypothetical async callbacks is not great in terms of programming, ideally the spawning function should run initially masked, to enable the possibility of doing some safety-critical initialization in OCaml.)

Reply:
Why would `Atom(0)` not crash but `Val_unit` would? Can you explain please? 

Also, can we add a code comment to state this explicitly please?

Reply:
Sure, `Atom(0)` is the empty array, and `Array.length` does not crash on it, whereas the latter crashes on input `Val_unit`.